src/cogs/events.py

```
@@ -0,0 +1,153 @@
+            if not events:
+                continue
+            
+            # Process oldest first (reverse of API response) to maintain narrative flow
+            for event in reversed(events):
+                if is_event_processed(event['id']):
+                    continue
+                
+                try:
+                    await self.process_event(channel, event, repo_url)
+                    mark_event_processed(event['id'])
+                except Exception as e:
```

Ruff (0.14.14)

[warning] Do not catch blind exception: `Exception`

(BLE001)